utils/doclint/check_public_api/JSBuilder.js

```diff
@@ -141,18 +146,46 @@ class JSOutline {
 }
 
 /**
+ * @param {!Array<!Documentation.Class>} classes
+ * @param {!Map<string, string>} inheritance
+ * @return {!Array<!Documentation.Class>}
+ */
+function recreateClassesWithInheritance(classes, inheritance) {
+  const classesByName = new Map(classes.map(cls => [cls.name, cls]));
+  return classes.map(cls => {
+    const membersMap = new Map();
+    for (let wp = cls; wp; wp = classesByName.get(inheritance.get(wp.name))) {
+      for (let member of wp.membersArray) {
+        // Member was overridden.
+        const memberId = member.type + ':' + member.name;
+        if (membersMap.has(memberId))
+          continue;
+        // Do not inherit constructors
+        if (wp !== cls && member.name === 'constructor' && member.type === 'method')
+          continue;
+        membersMap.set(memberId, member);
+      }
+    }
+    return new Documentation.Class(cls.name, Array.from(membersMap.values()));
+  });
+}
+
+/**
  * @param {!Array<!Source>} sources
  * @return {!Promise<{documentation: !Documentation, errors: !Array<string>}>}
  */
 module.exports = async function(sources) {
   const classes = [];
   const errors = [];
+  const inheritance = new Map();
   for (const source of sources) {
     const outline = new JSOutline(source.text());
     classes.push(...outline.classes);
     errors.push(...outline.errors);
+    for (let entry of outline.inheritance)
```

Comment:
Why are we cloning the map?

Reply:
that's because super classes are not necessarily defined in the same file